packages/web-components/src/calendar/calendar.ts

```diff
@@ -0,0 +1,813 @@
+   * @param event - 'dateselected' event
+   */
+  public secondaryCellSelectedHandler(event: any) {
+    const month = this.yearPickerOpen ? this.month : event.detail;
+    const year = this.yearPickerOpen ? event.detail : this.monthPickerYear;
+    if (this.yearPickerOpen) {
+      this.yearPickerOpen = false;
+    }
+    this.handleSwitchMonth(month, year);
+  }
+
+  /**
+   * Handles keyboard events on the navigation icons for both
+   * the date panel and the secondary panel
+   * @param event - Keyboard event
+   * @param panel - Panel to navigate on: primary or secondary
+   * @param direction - Direction of navigation: previous or next
+   */
+  public handleNavIconKeydown(event: KeyboardEvent, panel: string, direction: string): boolean {
+    if (event.key === keyEnter) {
+      if (panel === 'primary') {
+        if (direction === 'previous') {
+          this.handleSwitchMonth(this.getMonthInfo().previous.month, this.getMonthInfo().previous.year);
+        } else if (direction === 'next') {
+          this.handleSwitchMonth(this.getMonthInfo().next.month, this.getMonthInfo().next.year);
+        }
+      } else if (panel === 'secondary') {
+        if (direction === 'previous') {
+          this.yearPickerOpen
+            ? (this.yearPickerDecade = this.getYearPickerInfo().previousStart)
+            : (this.monthPickerYear = this.getMonthPickerInfo().previous);
+        } else if (direction === 'next') {
+          this.yearPickerOpen
+            ? (this.yearPickerDecade = this.getYearPickerInfo().nextStart)
+            : (this.monthPickerYear = this.getMonthPickerInfo().next);
+        }
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Handles keyboard events on the secondary panel title.
+   * @param event - Keyboard event
+   */
+  public handleSecondaryPanelTitleKeydown(event: KeyboardEvent): boolean {
+    if (event.key === keyEnter) {
+      this.toggleYearPicker();
+    }
+    return true;
+  }
+
+  /**
+   * Handles keyboard events on the "Go to today" link
+   * @param event - Keyboard event
+   */
+  public handleLinkKeydown(event: KeyboardEvent): boolean {
+    if (event.key === keyEnter) {
+      this.handleGoToToday();
+    }
+    return true;
+  }
+
+  /**
+   * Handles keyboard events on a day grid cell
+   * @param event - Keyboard event
+   * @param date - Date of the event target cell
+   */
+  public handleKeydown(event: KeyboardEvent, date: CalendarDateInfo): boolean {
+    super.handleKeydown(event, date);
+
+    const currentCell = event.target as HTMLElement;
+
+    // set navigatedDate to correspond to the date on the event target cell
+    this.navigatedDate = new Date(`${date.year}/${date.month}/${date.day}`);
+
+    switch (event.key) {
+      case keyArrowRight: {
+        event.preventDefault();
+        // Update navigatedDate
+        this.navigatedDate.setDate(date.day + 1);
+
+        // Update the month on the calendar if reached the end of the current month
```

Comment:
What does 7 and 1 represent? Would it make more sense if these were constants to increase readability and maintenance?

Reply:
They represent next day or next week. I can make a type for these in the options file!